### ISSUE TITLE
v5 component refactor: App Switcher (Top Bar)

### ISSUE THREAD

**Issue by nurm717123** [reapit/elements#222]
**Abstract**

_As part of the v5 Elements release, the ticket #153 was created. To break it into smaller part and allow for more focused discussions, this ticket has been created._

***app bar (parent)***
![image](https://github.com/user-attachments/assets/dbc50742-49e5-4c0d-aebf-2305aa72bfc8)

***this ticket***
![image](https://github.com/user-attachments/assets/abea3ff7-d037-4b0f-a890-8729a8de7980)

![image](https://github.com/user-attachments/assets/7c5e8c46-a585-48be-acd2-6f5306aa4dcc)

**Context**
- new app switcher not ready for dev yet
- would probably need separate discussion from top bar

As discussed in #153, this ticket involves creating a new app switcher based on the existing app switcher (`NavResponsiveAppSwitcher`) for the initial implementation. the app-switcher button will be using the v5 (new design) and the menu will be using the previous version.

**Specification**

- Is new component or existing? new
- Is the component being re-named? No
- Is the component design signed off by product and design? not yet, but become part of App Bar / Top Bar ticket
- [Guide to component structure & documentation in storybook](https://github.com/reapit/elements/wiki/Developer-Contribution-Guide)
- [Link to DS in Figma](https://www.figma.com/design/XJ6qcAV8gHscsUodqJMNEF/Reapit-Elements-production-ready-components?node-id=16-4749&m=dev)
- [Link to Elements Audit in Figma](https://www.figma.com/design/XJ6qcAV8gHscsUodqJMNEF/App-bar-and-Side-nav-%5BElements%5D?node-id=18-7578&t=AB2erFDHYuLLxvGj-4)
- [Accessibility requirements](https://m3.material.io/components/top-app-bar/accessibility)
- [Link to Design Guidelines](https://www.figma.com/design/pJ96bxoEFStjjIbAPy0Jaj/Design-Guidelines?node-id=945-25707&t=mvIlOCZjDYP90TZT-4)
- [Parent ticket](https://github.com/reapit/elements/issues/153)

**Developer Checklist**

- [ ] Styles alignment between Design System and Elements
- [ ] Check design tokens in Figma and implement CSS variable tokens if available for relevant component
- [ ] Align with accessibility standards / spec as per above
- [ ] If relevant, break down component into `Styles Only` and `React` component structures
- [ ] Ensure all variants of components are documented as appropriate 
- [ ] Ensure unit test coverage is adequate for component
- [ ] Update documentation in MDX file as per guidelines

**Release Checklist**

- [ ] Approved PR merged to main
- [ ] Design & product review and feedback addressed by developer
- [ ] Beta release by product / engineering lead to next beta version

**Additional Context or Information**

**Comment by kurtdoherty** [comment]
> creating a separate new app switcher based on the existing app switcher (v4) for the initial implementation.

**question:** what exactly do you mean by this?

**Comment by nurm717123** [comment]
Essentially, the new app switcher will feature an updated design, but the menu that opens when user click it will display the existing app switcher menu (`NavResponsiveAppSwitcher`).

**Comment by nurm717123** [comment]
my initial plan would be to make a new component so the consumer just need to call, .e.g

```javascript
<AppSwitcher options={[{label: 'AppMarket', onClick: handler}, ... /* and so on */]} />
```

and behind the scene it will look like, .e.g

```javascript
// app-switcher.tsx
<AppSwitcherContainer>
  <NavIconItem />
  <NavResponsiveAppSwitcher />
</AppSwitcherContainer>
```

**Comment by nurm717123** [comment]
yes, sure 👍

**Comment by nurm717123** [comment]
and also in the new design at mobile screen, the app switcher menu will move to the Mobile nav menu but the previous version is display under menu

![image](https://github.com/user-attachments/assets/6667c97d-179f-4056-a9ea-7c912a996a7b)

**Comment by nurm717123** [comment]
Following our previous discussion, the summary is 'The existing AppSwitcher and SideBar should still be usable with the new AppBar until they are replaced.' I will create a new ticket for this, so this ticket can serve as the new AppSwitcher once the design is ready for development.

**Comment by kurtdoherty** [comment]
Closing in favour of #428
